Gems/ROS2/Code/Source/SdfAssetBuilder/SdfAssetBuilderSettings.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <SdfAssetBuilder/SdfAssetBuilderSettings.h>
+
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/Settings/SettingsRegistryVisitorUtils.h>
+
+namespace ROS2
+{
+        namespace
+        {
+            constexpr const char* SdfAssetBuilderSupportedFileExtensionsRegistryKey = "/O3DE/ROS2/SdfAssetBuilder/SupportedFileTypeExtensions";
+            constexpr const char* SdfAssetBuilderUseArticulationsRegistryKey = "/O3DE/ROS2/SdfAssetBuilder/UseArticulations";
+        }
+
+    void SdfAssetBuilderSettings::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<SdfAssetBuilderSettings>()
+                ->Version(0)
+                ->Field("UseArticulations", &SdfAssetBuilderSettings::m_useArticulations)
+
+                // m_builderPatterns aren't serialized because we only use the serialization
+                // to detect when global settings changes cause us to rebuild our assets.
+                // A change to the builder patterns will cause the Asset Processor to add or
+                // remove the affected product assets, so we don't need to trigger any 
+                // additional rebuilds beyond that.
+             ;
+        }
+    }
+
+    void SdfAssetBuilderSettings::LoadGlobalSettings()
+    {
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        if (settingsRegistry == nullptr)
+        {
+            AZ_Error(SdfAssetBuilderName, false, "Settings Registry not found, Sdf Asset Builder settings will use defaults.");
+            return;
+        }
```

Comment:
Instead of this being only available for global settings, instead we can supply a `SettingsRegistryInterace*` pointer as a parameter and have it default to AZ::SettingsRegistry::Get().

If we do that, we can rename this function to `LoadSettings`
```suggestion
    void SdfAssetBuilderSettings::LoadGlobalSettings(AZ::SettingsRegistryInterface* settingsRegistry)
    {
        if (settingsRegistry == nullptr)
        {
            AZ_Error(SdfAssetBuilderName, false, "Settings Registry not found, Sdf Asset Builder settings will use defaults.");
            return;
        }
```